drivers/sensor/openrocket/openrocket_sensors.c

```diff
@@ -40,20 +44,31 @@ or_scalar_t or_lerp(or_scalar_t a, or_scalar_t b, or_scalar_t t) {
     return a * (1.0 - t) + b * t;
 }
 
-or_scalar_t or_get_time(unsigned int sampling_period_us, unsigned int lag_time_us) {
+or_scalar_t or_get_time(const struct or_common_params* cfg) {
     int64_t us = k_ticks_to_us_near64(k_uptime_ticks());
-    if (sampling_period_us != 0) {
-        us = (us / sampling_period_us) * sampling_period_us;
+    if (cfg->sampling_period_us != 0) {
+        us = (us / cfg->sampling_period_us) * cfg->sampling_period_us;
     }
-    us -= lag_time_us;
+    us -= cfg->lag_time_ms * 1000;
     us -= CONFIG_OPENROCKET_MS_BEFORE_LAUNCH * 1000;
     return ((or_scalar_t) (us)) / 1000000.0;
 }
 
+or_scalar_t or_random(or_scalar_t magnitude) {
+#ifdef CONFIG_OPENROCKET_NOISE
+    uint32_t r32 = sys_rand32_get();
```

Comment:
Is there a seed we can set to keep things deterministic if we were to integrate with CI eventually?

Reply:
Rn its whatever the system decides. now that you point it out it might be better to keep track of our own random state and have our own generator internal to openrocket subsystem so if another subsytem uses the system one it doesnt change

Reply:
Although, it would depend on what order sensors get measured in so if you add features that make the scheduler do something different enough it could change your random outputs